docs/source/notes/amp_examples.rst

```diff
@@ -0,0 +1,210 @@
+        torch.nn.utils.clip_grad_norm_(model.parameters(), max_norm*scaler.get_scale())
+
+        scaler.step(optimizer)
+        scaler.update()
+
+Here the *scaled* gradients are clipped.  ``scaler.step(optimizer)`` is aware that gradients
+have not yet been unscaled, and unscales them under the hood before calling ``optimizer.step()``.
+
+Gradient clipping with separate unscaling
+-----------------------------------------
+
+The specific case of clipping scaled gradients isn't so hard (all you have to do is clip to ``max_norm*scaler.get_scale()``).
+However, in general, between the backward pass and the optimizer step you may wish to manipulate gradients in some way that's not
+so easy to translate to scaled gradients.  In such cases, you can unscale and step separately.  Here's how that looks,
+using gradient clipping as an example once more::
+
+    scaler = AmpScaler()
+    ...
+    for input, target in data:
+        optimizer.zero_grad()
+        output = model(input)
+        loss = loss_fn(output, target)
+        scaler.scale(loss).backward()
+
+        scaler.unscale(optimizer)
+        # Since the optimizer's owned gradients are unscaled, we can clip to max_norm directly:
+        torch.nn.utils.clip_grad_norm_(model.parameters(), max_norm)
+
+        scaler.step(optimizer)
+        scaler.update()
+
+
+Gradient penalty
+----------------
+
+A gradient penalty loss term also requires awareness of the scale factor.
+Gradient penalty demonstrates some subtle/nonstandard use cases:
+
+* Creating out-of-place gradients with :func:`torch.autograd.grad`
+* Correct interaction of gradient scaling with double-backward
+
+Here's how that looks for a simple L2 penalty::
+
+    scaler = AmpScaler()
+    ...
+    for input, target in data:
+        optimizer.zero_grad()
+        output = model(input)
+        loss = loss_fn(output, target)
+
+        # We should scale outputs for the out-of-place backward pass
+        grad_params = torch.autograd.grad(scaler.scale(loss), model.parameters(), create_graph=True)
+
+        # In general, the penalty term may depend nonlinearly on the out-of-place gradients, so to be safe,
+        # manually unscale them before computing the penalty.  This unscale should be autograd-exposed.
+        grad_params = [p*(1./scaler.get_scale()) for p in grad_params]
+
+        # Compute the penalty term and add it to the loss
+        grad_norm = 0
+        for grad in grad_params:
+            grad_norm += grad.pow(2).sum()
+        grad_norm = grad_norm.sqrt()
+        loss = loss + grad_norm
+
+        # The usual scaling for backward will now accumulate leaf gradients that are appropriately scaled.
+        scaler.scale(loss).backward()
+        scaler.step(optimizer)
+        scaler.update()
+
+
+Gradient accumulation
+---------------------
+
+Gradient accumulation across iterations (between steps) is a common use case.
+:class:`AmpScaler` accommodates gradient accumulation without trouble::
```

Comment:
No need to comment on how troublesome (or not) a feature is. All your features should appear effortless to the user, after all.